class/class.midtrans-gateway-subscription.php

```diff
@@ -0,0 +1,225 @@
+    }
+
+    /**
+     * Admin Panel Options.
+     * HTML that will be displayed on Admin Panel.
+     * @access public
+     * @return void
+     */
+    public function admin_options() { ?>
+      <h3><?php _e( $this->pluginTitle(), 'midtrans-woocommerce' ); ?></h3>
+      <p><?php _e('Allows credit card subscription payments using Midtrans.', 'midtrans-woocommerce' ); ?></p>
+      <table class="form-table">
+        <?php
+          // Generate the HTML For the settings form.
+          $this->generate_settings_html();
+        ?>
+      </table><!--/.form-table-->
+      <?php
+    }
+      
+    /**
+     * initialize Gateway Settings Form Fields.
+     */
+    function init_form_fields() {
+      parent::init_form_fields();
+      WC_Midtrans_Utils::array_insert( $this->form_fields, 'enable_3d_secure', array(
+        'acquring_bank' => array(
+          'title' => __( 'Acquiring Bank', 'midtrans-woocommerce'),
+          'type' => 'text',
+          'label' => __( 'Acquiring Bank', 'midtrans-woocommerce' ),
+          'description' => __( 'Leave blank for default. </br> Specify your acquiring bank for this payment option. </br> Options: BCA, BRI, DANAMON, MAYBANK, BNI, MANDIRI, CIMB, etc (Only choose 1 bank).' , 'midtrans-woocommerce' ),
+          'default' => ''
+        ),
+        'bin_number' => array(
+          'title' => __( 'Allowed CC BINs', 'midtrans-woocommerce'),
+          'type' => 'text',
+          'label' => __( 'Allowed CC BINs', 'midtrans-woocommerce' ),
+          'description' => __( 'Leave this blank if you dont understand!</br> Fill with CC BIN numbers (or bank name) that you want to allow to use this payment button. </br> Separate BIN number with coma Example: 4,5,4811,bni,mandiri', 'midtrans-woocommerce' ),
+          'default' => ''
+        )
+      ));
+    }
+
+    /**
+     * This function auto-triggered by WC when payment process initiated.
+     * Serves as WC payment entry point.
+     * @param  [String] $order_id auto generated by WC.
+     * @return [array] contains redirect_url of payment for customer.
+     */
+    function process_payment( $order_id ) {
+      global $woocommerce;
+      
+      //create the order object.
+      $order = new WC_Order( $order_id );
+      // Get response object template.
+      $successResponse = $this->getResponseTemplate( $order );
+      // Get data for charge to midtrans API.
+      $params = $this->getPaymentRequestData( $order_id );
+
+      // add credit card payment.
+      $params['enabled_payments'] = ['credit_card'];
+      // add bank & channel migs params.
+      if (strlen($this->get_option('acquring_bank')) > 0)
+        $params['credit_card']['bank'] = strtoupper ($this->get_option('acquring_bank'));
+      // add bin params        
+      if (strlen($this->get_option('bin_number')) > 0){
+        $bins = explode(',', $this->get_option('bin_number'));
+        $params['credit_card']['whitelist_bins'] = $bins;
+      }
+
```

Comment:
On Snap payload, is it possible to please add `"custom_field3": "woocommerce-subscription-initial"` so it can be easily identified on Midtrans Dashboard.

Reply:
done on https://github.com/veritrans/SNAP-Woocommerce/pull/39/commits/ae189bdb9e8860c8a59516c82c338c4fea18a4e6